kernel/thread.c

```diff
@@ -946,8 +946,8 @@ void z_thread_mark_switched_out(void)
 #ifdef CONFIG_TRACING
 #ifdef CONFIG_THREAD_LOCAL_STORAGE
 	/* Dummy thread won't have TLS set up to run arbitrary code */
```

Comment:
what if we set up TLS for dummy threads?

Reply:
Dummy threads never run, so will never access TLS.  The idea is just a trick for re-using the context switch code for initial startup, you need some "thread" to "save" to when you want to start running your first thread but are still on a boot stack somewhere.